NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+    ForkThread(
+        function()
+            ScenarioFramework.EndOperation(ScenarioInfo.OpComplete, ScenarioInfo.OpComplete)
+        end
+    )
+end
+
+function M1NIS()
+	local Engineer1 = ScenarioInfo.UnitNames[Player1]['Engineer_1']
+    local Engineer2 = ScenarioInfo.UnitNames[Player1]['Engineer_2']
+    local Engineer3 = ScenarioInfo.UnitNames[Player1]['Engineer_3']
+    local Engineer4 = ScenarioInfo.UnitNames[Player1]['Engineer_4']
+
+	ScenarioFramework.CreateUnitDamagedTrigger(PlayerLose, ScenarioInfo.Ship, .75)
+
+	WaitSeconds(1)
+
+	Cinematics.EnterNISMode()
+
+	ScenarioFramework.Dialogue(OpStrings.IntroNIS_Dialogue, nil, true)
+	Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_01a'), 0)
+	WaitSeconds(.5)
+	Cinematics.CameraTrackEntity(Engineer1, 40, 3)
+
+	WaitSeconds(7)
+
+	Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_01b'), 2)
+
+	Cinematics.ExitNISMode()
+
+    if ScenarioInfo.HumanPlayers == 2 then
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer2, Player2)
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer3, Player2)
+    elseif ScenarioInfo.HumanPlayers == 3 then
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer2, Player2)
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer3, Player3)
+    elseif ScenarioInfo.HumanPlayers == 4 then
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer2, Player2)
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer3, Player3)
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer4, Player4)
+    end
+
+	WaitSeconds(1)
+	ForkThread(M1)
+end
+
+
+function M1()
+	ScenarioFramework.CreateTimerTrigger(M1StartUEFScouting, UEFM1Timer[Difficulty])
+
+	local function MissionNameAnnouncement()
+        ScenarioFramework.SimAnnouncement(ScenarioInfo.name, 'mission by [e]JJs_AI')
+    end
+
+    ScenarioFramework.CreateTimerTrigger(MissionNameAnnouncement, 7)
+
+    ScenarioInfo.M1P1 = Objectives.Protect(
+        'primary',                    -- type
+        'incomplete',                   -- complete
+        'Defend Command Ship',  -- title
+        'Defend the Command Ship at all costs. We need it to leave this planet.',  -- description
+        {                               -- target
+            Units = {ScenarioInfo.Ship},
+        }
+    )
+    table.insert(AssignedObjectives, ScenarioInfo.M1P1)
+
+	ScenarioFramework.CreateArmyIntelTrigger(ForkAttackThread, ArmyBrains[UEF], 'LOSNow', false, true,  categories.NOMADS, true, ArmyBrains[Player1] )
+end
+
+function M1StartUEFScouting()
+	----------
+	-- Scouting Starts
+	----------
+	local Scouts = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'Scouts','NoFormation')
+
+	for k, v in Scouts:GetPlatoonUnits() do
+    	ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M1_UEF_Scout_Chain')))
+    end
+end
+
+function ForkAttackThread()
+	ForkThread(M1StartUEFAttacks)

```

Comment:
you should not need to use ForkThread here since this function is called by a trigger, which is already running in a thread
So setting `M1StartUEFAttacks` as cb function of the trigger should work